### PR TITLE
(maint) Install rugged gem on puppetserver container

### PR DESCRIPTION
**Pull request by underscorgan**
This is needed for the `rugged` r10k backend.

### CODE REVIEW

File: docker/puppetserver-standalone/Dockerfile [hunk 1]
```diff
@@ -18,6 +31,29 @@ RUN apt-get update && \
     /usr/local/bin/lein && \
     gem install --no-doc bundler fpm
 
+RUN wget http://nightlies.puppet.com/apt/puppet6-nightly-release-"$UBUNTU_CODENAME".deb && \
+    dpkg -i puppet6-nightly-release-"$UBUNTU_CODENAME".deb && \
+    rm puppet6-nightly-release-"$UBUNTU_CODENAME".deb && \
+    apt-get update && \
+    apt-get install --no-install-recommends puppet-agent && \
+    git clone --single-branch --branch native_fix https://github.com/underscorgan/rake-compiler && \
```
Comment (Iristyle) [inline docker/puppetserver-standalone/Dockerfile]: Do we want a note about why we're using a fork / should we move this over to `puppetlabs` org?

Comment (underscorgan) [reply docker/puppetserver-standalone/Dockerfile]: I have a note down a couple of lines. I got a comment on my PR yesterday so I'm holding out hope that we'll be able to not use a fork, but if it lingers too long can move it to the puppetlabs org

Comment (Iristyle) [reply docker/puppetserver-standalone/Dockerfile]: Ah sorry, totally missed that.

See also https://github.com/rake-compiler/rake-compiler/pull/160

File: docker/puppetserver-standalone/Dockerfile [hunk 2]
```diff
@@ -18,6 +31,29 @@ RUN apt-get update && \
     /usr/local/bin/lein && \
     gem install --no-doc bundler fpm
 
+RUN wget http://nightlies.puppet.com/apt/puppet6-nightly-release-"$UBUNTU_CODENAME".deb && \
+    dpkg -i puppet6-nightly-release-"$UBUNTU_CODENAME".deb && \
+    rm puppet6-nightly-release-"$UBUNTU_CODENAME".deb && \
+    apt-get update && \
+    apt-get install --no-install-recommends puppet-agent && \
```
Comment (Iristyle) [inline docker/puppetserver-standalone/Dockerfile]: Looks like we need the `-y` here to appease lint

**Comment by underscorgan** [comment]
@Iristyle updated!

File: docker/puppetserver-standalone/Dockerfile (hunk 1 above)
Comment (Iristyle) [reply docker/puppetserver-standalone/Dockerfile]: So the latest comment at https://github.com/rake-compiler/rake-compiler/pull/160#issuecomment-560029680 says to just `gem install --no-ri --no-rdoc rugged` like you would normally... and then copy the generated gem out of `GEM_HOME` from the build container.

That way, no `rake-compiler` is ever needed.

**Comment by underscorgan** [comment]
ugh, so I started working on that and found this PR was missing a build-flag so the rugged gem isn't linking against thing like curl and libgit, which I think will probably make this not actually useful (and ubuntu doesn't seem to have compatible version of libgit). I'm going to close this for now, this one's going to be rather a lot of work.

**Pull request closed without merging**